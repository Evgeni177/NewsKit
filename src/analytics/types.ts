export type TealiumLinkEvent = {
  event_navigation_action: React.ReactNode;
  event_navigation_browsing_method?: string;
};

export type TealiumViewEvent = {
  page_url: string;
};

export type InteractionEvent = {
  type: 'link';
  data: TealiumLinkEvent;
};

export type PageViewEvent = {
  type: 'view';
  data: TealiumViewEvent;
};

export type AnalyticsEvent = InteractionEvent | PageViewEvent;

export type UTag = {
  link(e: TealiumLinkEvent): void;
  view(e: TealiumViewEvent): void;
};

export interface ExtendedWindow extends Window {
  utag: UTag;
}
