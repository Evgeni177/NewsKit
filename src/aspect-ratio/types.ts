import {
  MaxHeightProperty,
  MaxWidthProperty,
  PaddingTopProperty,
  HeightProperty,
  WidthProperty,
} from 'csstype';
import {GetAspectRatioProp} from '../utils/get-aspect-ratio';

export interface ContainerProp {
  paddingTop?: PaddingTopProperty<string>;
  maxHeight?: MaxHeightProperty<string>;
  maxWidth?: MaxWidthProperty<string>;
}

export interface StyledContainerProps {
  $height?: HeightProperty<string>;
  $width?: WidthProperty<string>;
}

export interface AspectRatioProps
  extends GetAspectRatioProp,
    Omit<ContainerProp, 'paddingTop'> {}
