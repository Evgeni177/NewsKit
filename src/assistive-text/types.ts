import {EnhancerOverrides} from '../form/types';
import {MQ} from '../utils/style';
import {WithEnhancersProps} from '../with-enhancers/types';

export type AssistiveTextProps = React.HTMLAttributes<HTMLParagraphElement> &
  Omit<
    WithEnhancersProps,
    'enhancersType' | 'isFocused' | 'componentDefaultsPath'
  > & {
    overrides?: EnhancerOverrides & {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
      minHeight?: MQ<string>;
    };
  };
