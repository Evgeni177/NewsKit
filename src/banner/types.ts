import {NewsKitIconProps} from '../icons';
import {MQ} from '../utils/style';

export interface BannerBaseProps extends React.AriaAttributes {
  actions?: React.ComponentType[];
  children: Exclude<React.ReactNode, 'undefined'>;
  icon?: React.ReactElement<NewsKitIconProps>;
  closeButtonLabel?: string;
  onClose?: () => void;
  overrides?: {
    stylePreset?: MQ<string>;
    minHeight?: MQ<string>;
    spaceInset?: MQ<string>;
    maxWidth?: MQ<string>;
    icon?: {
      spaceInline?: MQ<string>;
    };
    content?: {
      spaceInline?: MQ<string>;
      message?: {
        stylePreset?: MQ<string>;
        typographyPreset?: MQ<string>;
      };
    };
    actions?: {
      spaceInline?: MQ<string>;
      closeButton?: {
        stylePreset?: MQ<string>;
      };
    };
  };
}

export interface BannerInternalProps extends BannerBaseProps {
  layout: 'horizontal' | 'vertical';
}

export interface BannerProps extends BannerBaseProps {
  layout?: MQ<'horizontal' | 'vertical'>;
}
