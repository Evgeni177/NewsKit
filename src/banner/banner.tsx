import React from 'react';
import {BannerProps} from './types';
import {
  StyledBannerContainer,
  StyledInnerContainer,
  StyledContentContainer,
  StyledIconContainer,
  StyledMessageContainer,
} from './styled';

export const Banner: React.FC<BannerProps> = ({
  overrides,
  children,
  icon,
  ...restProps
}) => (
  <StyledBannerContainer
    role="region"
    aria-live="polite"
    data-testid="banner-container"
    overrides={overrides}
    {...restProps}
  >
    <StyledInnerContainer
      flow="horizontal-top"
      stackDistribution="flex-start"
      wrap="nowrap"
      overrides={overrides}
    >
      {icon && (
        <StyledIconContainer overrides={overrides}>{icon}</StyledIconContainer>
      )}
      <StyledContentContainer overrides={overrides}>
        <StyledMessageContainer
          overrides={overrides}
          as={
            typeof children === 'string' ||
            (Array.isArray(children) &&
              children.some(child => typeof child === 'string'))
              ? 'p'
              : 'div'
          }
        >
          {children}
        </StyledMessageContainer>
      </StyledContentContainer>
    </StyledInnerContainer>
  </StyledBannerContainer>
);
