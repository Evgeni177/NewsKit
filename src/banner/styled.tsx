import {BannerProps} from './types';
import {
  styled,
  getStylePreset,
  getResponsiveSize,
  getResponsiveSpace,
  getTypographyPreset,
} from '../utils/style';
import {Stack} from '../stack';
import {TextBlock} from '../text-block';

export const StyledBannerContainer = styled.div<BannerProps>`
  box-sizing: border-box;
  ${getStylePreset('banner', '')};
  ${getResponsiveSpace('padding', 'banner', '', 'spaceInset')}
  ${getResponsiveSize('minHeight', 'banner', '', 'minHeight')}
`;

export const StyledInnerContainer = styled(Stack)<
  Pick<BannerProps, 'overrides'>
>`
  box-sizing: border-box;
  ${getResponsiveSize(
    'maxWidth',
    'banner.innerContainer',
    'innerContainer',
    'maxWidth',
  )}
  margin: 0 auto;
`;

export const StyledIconContainer = styled.div<Pick<BannerProps, 'overrides'>>`
  ${getResponsiveSpace(
    'marginRight',
    'banner.innerContainer.icon',
    'innerContainer.icon',
    'spaceInline',
  )}
`;

export const StyledContentContainer = styled.div<
  Pick<BannerProps, 'overrides'>
>`
  ${getResponsiveSpace(
    'marginRight',
    'banner.innerContainer.content',
    'innerContainer.content',
    'spaceInline',
  )}
`;

export const StyledMessageContainer = styled(TextBlock)<
  Pick<BannerProps, 'overrides'>
>`
  ${getTypographyPreset(
    'banner.innerContainer.content.message',
    'innerContainer.content.message',
    {
      withCrop: true,
    },
  )};

  ${getStylePreset(
    'banner.innerContainer.content.message',
    'innerContainer.content.message',
  )};
`;
