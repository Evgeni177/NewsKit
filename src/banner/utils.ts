import {BreakpointKeys, Theme} from '../theme';
import {MQ} from '../utils';
import {sortBreakpointKeys} from '../utils/hooks/use-media-query/utils';
import {BannerInternalProps} from './types';

const isVertical = (layout?: BannerInternalProps['layout']) =>
  layout === 'vertical';

export const getVisibleBreakpointsForLayout = (
  layout: Partial<
    {
      [key in BreakpointKeys]: 'horizontal' | 'vertical';
    }
  >,
  theme: Theme,
) => {
  const horizontalBreakpoints: MQ<boolean> = {};
  const verticalBreakpoints: MQ<boolean> = {};

  const {breakpoints} = theme;
  const breakpointsKeys = Object.keys(breakpoints) as BreakpointKeys[];

  // Get the lowest breakpoint key as base
  const baseBreakpoint: BreakpointKeys = sortBreakpointKeys(
    Object.keys(layout) as BreakpointKeys[],
    breakpointsKeys,
  )[0];
  let baseValue = layout[baseBreakpoint];

  // Set values to all breakpoints (mainly for the missing ones)
  const valuesForAllBreakpoints = breakpointsKeys.reduce(
    (acc, breakpointKey) => {
      if (acc[breakpointKey] === undefined) {
        return {...acc, [breakpointKey]: baseValue};
      }

      baseValue = acc[breakpointKey];
      return acc;
    },
    layout,
  );

  const valuesForAllBreakpointsAsArr = Object.entries(valuesForAllBreakpoints);

  // Creating breakpoint objects like:
  // verticalBreakpoints = {
  //   xs: true,
  //   sm: true,
  //   md: false,
  //   lg: false,
  //   xl: false,
  // }
  // That can be passed to the visible component based on the layout
  valuesForAllBreakpointsAsArr.forEach(([key, value]) => {
    verticalBreakpoints[key as BreakpointKeys] = isVertical(value);
    horizontalBreakpoints[key as BreakpointKeys] = !isVertical(value);
  });

  return {
    verticalBreakpoints,
    horizontalBreakpoints,
  };
};
