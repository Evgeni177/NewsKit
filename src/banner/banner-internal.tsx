import React from 'react';
import {BannerInternalProps} from './types';
import {
  StyledBannerContainer,
  StyledMaxWidthContainer,
  StyledContentContainer,
  StyledIconContainer,
  StyledMessageContainer,
  StyledActionsContainer,
  StyledIconContentContainer,
} from './styled';
import {renderIfReactComponent} from '../utils/component';
import {Flow, StackDistribution} from '../stack';
import {AlignSelfValues, StackChild} from '../stack-child';
import {useTheme} from '../theme';
import {useReactKeys} from '../utils/hooks/use-react-keys';

export const BannerInternal: React.FC<BannerInternalProps> = ({
  actions,
  children,
  icon,
  overrides,
  layout,
  ...restProps
}) => {
  const theme = useTheme();

  const actionsCount = actions ? actions.length : 0;
  const actionKeys = useReactKeys(actionsCount);

  const actionsSpacing =
    overrides?.actions?.spaceInline ||
    theme.componentDefaults.banner[layout].actions.spaceInline;

  const hasActions = Array.isArray(actions) && actionKeys.length;

  return (
    <StyledBannerContainer
      role="region"
      data-testid="banner-container"
      overrides={overrides}
      layout={layout}
      {...restProps}
    >
      <StyledMaxWidthContainer
        flow={
          layout === 'vertical' ? Flow.VerticalCenter : Flow.HorizontalCenter
        }
        stackDistribution="flex-start"
        wrap="nowrap"
        layout={layout}
        overrides={overrides}
      >
        <StyledIconContentContainer
          flow={Flow.HorizontalTop}
          stackDistribution="flex-start"
          wrap="nowrap"
          layout={layout}
          overrides={overrides}
        >
          {icon && (
            <StyledIconContainer layout={layout} overrides={overrides}>
              {icon}
            </StyledIconContainer>
          )}
          <StyledContentContainer>
            <StyledMessageContainer
              layout={layout}
              overrides={overrides}
              as={
                typeof children === 'string' ||
                (Array.isArray(children) &&
                  children.some(child => typeof child === 'string'))
                  ? 'p'
                  : 'div'
              }
            >
              {children}
            </StyledMessageContainer>
          </StyledContentContainer>
        </StyledIconContentContainer>
        {hasActions && (
          <StyledActionsContainer
            flow={
              layout === 'vertical'
                ? Flow.VerticalCenter
                : Flow.HorizontalCenter
            }
            stackDistribution={StackDistribution.Start}
            spaceInline={actionsSpacing}
            layout={layout}
          >
            {actions &&
              actions.map((action, idx) => (
                <StackChild
                  key={actionKeys[idx]}
                  alignSelf={
                    layout === 'vertical'
                      ? AlignSelfValues.Stretch
                      : AlignSelfValues.Auto
                  }
                >
                  {renderIfReactComponent(action)}
                </StackChild>
              ))}
          </StyledActionsContainer>
        )}
      </StyledMaxWidthContainer>
    </StyledBannerContainer>
  );
};
