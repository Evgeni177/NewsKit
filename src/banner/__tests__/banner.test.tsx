import React from 'react';
import {Banner} from '../banner';
import {BannerProps} from '../types';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {TextBlock} from '../../text-block';
import {IconFilledInfo} from '../../icons';
import {Link} from '../../link';

const bannerMessage = 'Banner text';
const bannerIcon = (
  <IconFilledInfo
    overrides={{size: 'iconSize020', stylePreset: 'inkInverse'}}
  />
);

describe('Banner', () => {
  test('renders with default props', () => {
    const props: BannerProps = {
      children: bannerMessage,
    };
    const fragment = renderToFragmentWithTheme(Banner, props) as any;

    expect(fragment).toMatchSnapshot();
  });
  test('renders with icon', () => {
    const props: BannerProps = {
      children: bannerMessage,
      icon: bannerIcon,
    };
    const fragment = renderToFragmentWithTheme(Banner, props) as any;

    expect(fragment).toMatchSnapshot();
  });
  test('renders with content as react element', () => {
    const props: BannerProps = {
      children: <TextBlock>{bannerMessage}</TextBlock>,
    };
    const fragment = renderToFragmentWithTheme(Banner, props) as any;

    expect(fragment).toMatchSnapshot();
  });
  test('renders with content as string and link', () => {
    const props: BannerProps = {
      children: [
        `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt`,
        <Link href="/">NewsKit Link</Link>,
        `ut labore et dolore magna aliqua.`,
      ],
    };
    const fragment = renderToFragmentWithTheme(Banner, props) as any;

    expect(fragment).toMatchSnapshot();
  });
});
