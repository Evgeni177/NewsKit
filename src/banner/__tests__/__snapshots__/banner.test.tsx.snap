// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders with content as react element 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  background-color: #128293;
  padding: 20px;
  min-height: 64px;
}

.emotion-0 svg {
  fill: #ffffff;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  letter-spacing: 0;
  color: #ffffff;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-3 svg {
  fill: #ffffff;
}

.emotion-4 {
  margin: 0;
  padding: 1px 0;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-testid="banner-container"
    role="region"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <p
            class="emotion-4"
          >
            Banner text
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner renders with content as string and link 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  background-color: #128293;
  padding: 20px;
  min-height: 64px;
}

.emotion-0 svg {
  fill: #ffffff;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  letter-spacing: 0;
  color: #ffffff;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-3 svg {
  fill: #ffffff;
}

.emotion-4 {
  display: inline-block;
  transition-property: color;
  transition-duration: 200ms;
  transition-timing-function: cubic-bezier(0, 0, .5, 1);
  color: #0a68c1;
}

.emotion-4 svg {
  fill: #0a68c1;
}

.emotion-4:hover:not(:disabled) {
  color: #02509b;
}

.emotion-4:hover:not(:disabled) svg {
  fill: #02509b;
}

.emotion-4:active:not(:disabled) {
  color: #01396e;
}

.emotion-4:active:not(:disabled) svg {
  fill: #01396e;
}

.emotion-4:visited:not(:disabled) {
  color: #7547a3;
}

.emotion-4:visited:not(:disabled) svg {
  fill: #7547a3;
}

@media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
  .emotion-4 {
    vertical-align: text-bottom;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-6 {
  margin: 0;
  padding: 1px 0;
  display: inline-block;
  display: block;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-testid="banner-container"
    role="region"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-3"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
      eiusmod tempor incididunt
          <a
            class="emotion-4"
            href="/"
          >
            <span
              class="emotion-5"
            >
              <span
                class="emotion-6"
              >
                NewsKit Link
              </span>
            </span>
          </a>
          ut labore et dolore magna aliqua.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner renders with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  background-color: #128293;
  padding: 20px;
  min-height: 64px;
}

.emotion-0 svg {
  fill: #ffffff;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-3 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  letter-spacing: 0;
  color: #ffffff;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-3 svg {
  fill: #ffffff;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-testid="banner-container"
    role="region"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-3"
        >
          Banner text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner renders with icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  background-color: #128293;
  padding: 20px;
  min-height: 64px;
}

.emotion-0 svg {
  fill: #ffffff;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  box-sizing: border-box;
  max-width: 1920px;
  margin: 0 auto;
}

.emotion-2 {
  margin-right: 12px;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  color: #ffffff;
  fill: #ffffff;
  vertical-align: unset;
  display: inline-block;
}

.emotion-3.emotion-3 {
  width: 24px;
  height: 24px;
  color: #ffffff;
  fill: #ffffff;
}

.emotion-6 {
  margin: 0;
  padding: 1px 0;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  letter-spacing: 0;
  color: #ffffff;
}

.emotion-6::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-6::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

.emotion-6 svg {
  fill: #ffffff;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-testid="banner-container"
    role="region"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <svg
          aria-hidden="true"
          class="emotion-3 emotion-4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          />
        </svg>
      </div>
      <div
        class="emotion-2"
      >
        <p
          class="emotion-6"
        >
          Banner text
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
